python/pyspark/worker.py

```diff
@@ -178,7 +178,7 @@ def process():
     except Exception:
         try:
             write_int(SpecialLengths.PYTHON_EXCEPTION_THROWN, outfile)
-            write_with_length(traceback.format_exc().encode("utf-8"), outfile)
+            write_with_length(traceback.format_exc(), outfile)
```

Comment:
Yes, It can work with Python2 and 3+.
In Python 3+,  has encode("utf-8") or not is all ok.
In Python 2, should remove encode("utf-8"). 
Or we can use try except to slove it like [#18262](https://github.com/apache/spark/pull/18262)